lib/client-options.js

```diff
@@ -68,6 +69,7 @@ function defaultOptions () {
     },
     authProvider: null,
     requestTracker: null,
+    metrics: new metrics.DefaultMetrics(),
```

Comment:
Do we expect much of an added performance cost for emitting events with the `DefaultMetrics` implementation?  I imagine at least initially, users will not use metrics, so if emitting does have a performance cost, maybe we default to a `no-op` implementation where the `onXX` methods don't do anything.

Reply:
The cost of emitting events is only the function call, [when there are no listeners to the event it can be considered no-op](https://github.com/nodejs/node/blob/v10.13.0/lib/events.js#L176-L179).